cluster/v1alpha1/types.go

```diff
@@ -0,0 +1,43 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:resource:scope="Namespaced"
+
+// ManagedClusterSetBinding projects a ManagedClusterSet into a certain namespace.
+// User is able to create a ManagedClusterSetBinding in a namespace and bind it to a
+// ManagedClusterSet if they have an RBAC rule to GET on the virtual subresource of
+// managedclustersets/bind. Workloads created in the same namespace can only be
+// distributed to ManagedClusters in ManagedClustersets bound in this namespace by
+// higher level controllers.
+type ManagedClusterSetBinding struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines the attributes of ManagedClusterSetBinding.
+	Spec ManagedClusterSetBindingSpec `json:"spec"`
+}
+
+// ManagedClusterSetBindingSpec defines the attributes of ManagedClusterSetBinding.
+type ManagedClusterSetBindingSpec struct {
+	// ClusterSet is the name of the ManagedClusterSet to bind.
```

Comment:
you need more doc around what dangling references do, whether there are any native restrictions against the clusterset name (can anyone bind any clusterset).  Whether instance names are required to match the clusterset name.

Reply:
Doc added